### PR TITLE
fix(tag): add title to tags

### LINKED ISSUE #6111: Tag component only sets title if filter
I have a Tag that contains long text, so it is truncated with an ellipsis.  I want to set the title attribute so that when the user hovers over the tag, they can see the full text.  

In "carbon-components-react": "7.9.0", if the tag is a filter tag, the title gets passed through.  Otherwise, the title is not used.  I don't want to use the filter just to get the title because it requires a lot of styling overrides.

In the latest code, title doesn't work for the filter version, either.

### PR DESCRIPTION
**Pull request by tw15egan**
Closes https://github.com/carbon-design-system/carbon/issues/6111

The `title` attribute on `Tag` allows the user to customize to text that appears on the `aria-label` for the close icon. This PR adds a boolean prop, `enableTagTitle`, that when enabled takes the value of the tag and will display it on hover.

#### Changelog

**New**

- `enableTagTitle` prop to allow for a `title` attribute to be added on the `Tag`

**Changed**
- `title` now is added to the close icon, if provided. Previously was only being added as an aria-label, which due to the naming seems confusing. 
- Text updates to the `Tag` story

#### Testing / Reviewing

Enable/disable the new prop, ensure it works properly

### CODE REVIEW

**Comment by joshblack** [comment]
@tw15egan would we want to set the `title` by default, or nah?

**Review by jeanservaas — APPROVED** [review]

**Comment by tw15egan** [comment]
@joshblack doesn't make much of a difference to me, figured it would be best to opt into this feature than to make it default

**Comment by emyarod** [comment]
I'm not sure if there are additional a11y reasons for only setting the `title` to filter tags (#4163, #4863) but if we do want to apply it to non filter tags (and it doesn't have any a11y impact), I think it would be better to keep the API surface area low and set it by default. it seems there is a workaround already according to the original issue author so just want to double check why `title` is specific to filter tags at the moment. my guess is because we don't encourage truncating tags by default?

File: packages/react/src/components/Tag/Tag.js [hunk 1]
```diff
@@ -61,19 +62,25 @@ const Tag = ({
       }
       id={tagId}
       {...other}>
-      <span className={`${prefix}--tag__label`}>
+      <span
+        className={`${prefix}--tag__label`}
+        title={enableTagTitle ? children : null}>
```
Comment (asudoh) [inline packages/react/src/components/Tag/Tag.js]:
```suggestion
        title={enableTagTitle && typeof children === 'string' ? children : null}>
```

File: packages/react/src/components/Tag/Tag.js [hunk 2]
```diff
@@ -61,19 +62,25 @@ const Tag = ({
...
+        aria-labelledby={tagId}
+        title={title}>
         <Close16 />
       </button>
     </div>
   ) : (
-    <span className={tagClasses} {...other}>
+    <span
+      className={tagClasses}
+      title={enableTagTitle ? children : null}
```
Comment (asudoh) [inline packages/react/src/components/Tag/Tag.js]:
```suggestion
      title={enableTagTitle && typeof children === 'string' ? children : null}
```

**Review by asudoh — COMMENTED** [review]
LGTM basically, assuming that we have a condition where doesn't want the title hover. Thanks @tw15egan!

**Comment by tw15egan** [comment]
@emyarod updated it to be enabled by default 👍 

@asudoh added your changes 👍

**Review by emyarod — COMMENTED** [review]
does there need to be a prop for enabling the title? it should just automatically take the prop value right?

**Comment by tw15egan** [comment]
@emyarod do we want it always to be enabled with no way to remove it? I was just following convention like what we do with `MultiSelect` and `useTitleInItem`

**Comment by emyarod** [comment]
for multiselect I believe that prop is originally from a few years ago and more recently we have been setting titles unconditionally (https://github.com/carbon-design-system/carbon/pull/5621/files https://github.com/carbon-design-system/carbon/pull/5102). but I think the context for those components is slightly different from tags

what I meant in my earlier comment (https://github.com/carbon-design-system/carbon/pull/6112#issuecomment-632193921) was I think it makes more sense for the title logic to be identical (or similar) regardless of whether or not it is a filter tag. but if it makes sense to conditionally set the title only for non-filter tags then the new prop is fine

**Comment by tw15egan** [comment]
I'm good with that, I'll just set it by default and remove the prop

File: packages/react/src/components/Tag/Tag-story.js [hunk 3]
```diff
@@ -61,7 +62,7 @@ storiesOf('Tag', module)
     'Filter',
     () => (
       <Tag className="some-class" {...props.filter()} filter>
-        {text('Content (children)', 'This is not a tag')}
+        {text('Content (children)', 'This is a tag')}
```
Comment (dakahn) [inline packages/react/src/components/Tag/Tag-story.js:64]: Thank you for changing this :D

**Review by dakahn — APPROVED** [review]

**Review by asudoh — COMMENTED** [review]
LGTM 👍, wanted to have @emyarod's final review. Thanks @tw15egan for the update!

**Review by emyarod — APPROVED** [review]
looks good to me

**Review by asudoh — APPROVED** [review]
🎉

**Pull request merged**